107/src/main/java/org/ehcache/jsr107/EhcacheCachingProvider.java

```diff
@@ -147,10 +148,10 @@ private Eh107CacheManager createCacheManager(URI uri, Configuration config, Prop
       services.add(new DefaultJsr107SerializationProvider());
     }
 
-    Eh107InternalCacheManager ehcacheManager = new Eh107InternalCacheManager(config, services, !jsr107Service.jsr107CompliantAtomics());
+    org.ehcache.CacheManager ehcacheManager = new EhcacheManager(config, services, !jsr107Service.jsr107CompliantAtomics());
     ehcacheManager.init();
 
-    return new Eh107CacheManager(this, ehcacheManager, properties, config.getClassLoader(), uri,
+    return new Eh107CacheManager(this, ehcacheManager, jsr107Service.getStatistics(), properties, config.getClassLoader(), uri,
```

Comment:
It still feels weird to pass the `StatisticsService` since the `ehcacheManager` is passed and already has it.

Reply:
Why not passing jsr107service and retrieve the statistics inside the constructor? I would prefer that.